### DISCUSSION TITLE
Can someone explain the wrapping of jars into bundles?

### DISCUSSION THREAD

**Discussion by hammyau** [eclipse-tycho/tycho#499] [Q&A] [answered]
Does this provide a means of wrapping jars that can then be used in a build? 
That is so they appear to be bundles?

In my target file I have
```
	    <location includeDependencyScope="compile" includeSource="false" missingManifest="generate" type="Maven">
		    <dependencies>
			    <dependency>
				    <groupId>org.mine</groupId>
				    <artifactId>blah</artifactId>
				    <version>1.0-SNAPSHOT</version>
				    <type>jar</type>
			    </dependency>
           ...

```
This seems to give me the wrapped jar.

How then are they referred to and how is the jar added to the runtime classpath? 
My product has a comforting series of green SUCCESS messages.
But at runtime it fails to load the relevant classes.

**Comment by laeubi** [comment, marked as answer]
Most probably your jar is missing some dependencies, the default wrapping adds everything as optional.
By the way, if it is really "yours" as indicated by the `org.mine` you should consider adding OSGi metadata in the first place.

If you want to see whats the result of the whole thing, tick the "show location content" checkbox in the target editor.

**Reply by hammyau** [reply]
Thank you Christoph, that pointed me in the correct direction.
Sadly in this world of computing I/we still see what we want to see.  
Not what is there. A simple addition of .0 to the version was enough. :-)

So my build now works. But I suspect not in the correct way.
I think I will work on some improvements. And change my libraries to plugins.
In the end I just use external jars not wrapped at all. Simply placed in a lib directory.
